cmd/disk-cache.go

```diff
@@ -649,9 +684,68 @@ func (c *cacheObjects) PutObject(ctx context.Context, bucket, object string, r *
 			}
 		}()
 	}
-
 	return objInfo, err
+}
+
+// upload cached object to backend in async commit mode.
+func (c *cacheObjects) uploadObject(ctx context.Context, oi ObjectInfo) {
```

Comment:
What keeps 2 of these from running concurrently on the same object or a delete taking place on the object concurrently?

Reply:
there is a lock on the cache path, which should prevent concurrent uploads/deletes